pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/properties/AbstractMultiNumericProperty.java

```diff
@@ -4,48 +4,103 @@
+import java.util.Map;
+
+import net.sourceforge.pmd.NumericPropertyDescriptor;
+import net.sourceforge.pmd.PropertyDescriptorField;
+
 /**
+ * Base class for multi-valued numeric properties.
+ *
+ * @param <T> The type of number
  *
  * @author Brian Remedios
- * @param <T>
+ * @version Refactored June 2017 (6.0.0)
  */
-public abstract class AbstractMultiNumericProperty<T> extends AbstractNumericProperty<T> {
+/* default */ abstract class AbstractMultiNumericProperty<T extends Number> extends AbstractMultiValueProperty<T>
+    implements NumericPropertyDescriptor<List<T>> {
+
+    private final T lowerLimit;
+    private final T upperLimit;
+
 
     /**
-     * Constructor for AbstractMultiNumericProperty.
+     * Constructor for a multi-valued numeric property using a list of defaults.
+     *
+     * @param theName        Name
+     * @param theDescription Description
+     * @param lower          Minimum value of the property
+     * @param upper          Maximum value of the property
+     * @param theDefault     List of defaults
+     * @param theUIOrder     UI order
      *
-     * @param theName
-     *            String
-     * @param theDescription
-     *            String
-     * @param lower
-     *            Number
-     * @param upper
-     *            Number
-     * @param theDefault
-     *            T
-     * @param theUIOrder
-     *            float
+     * @throws IllegalArgumentException if lower > upper, or one of them is null, or one of the defaults is not between
+     * the bounds
      */
-    protected AbstractMultiNumericProperty(String theName, String theDescription, Number lower, Number upper,
-            T theDefault, float theUIOrder) {
-        super(theName, theDescription, lower, upper, theDefault, theUIOrder);
+    AbstractMultiNumericProperty(String theName, String theDescription, T lower, T upper, List<T> theDefault,
+                                 float theUIOrder) {
+        super(theName, theDescription, theDefault, theUIOrder);
+
+        lowerLimit = lower;
+        upperLimit = upper;
+
+        checkNumber(lower);
+        checkNumber(upper);
+
+        if (lower.doubleValue() > upper.doubleValue()) {
+            throw new IllegalArgumentException("Lower limit cannot be greater than the upper limit");
+        }
+
+
     }
 
-    /**
-     * @return boolean
-     * @see net.sourceforge.pmd.PropertyDescriptor#isMultiValue()
-     */
+
+    private void checkNumber(T number) {
+        if (valueErrorFor(number) != null) {
+            throw new IllegalArgumentException(valueErrorFor(number));
+        }
+    }
+
+
     @Override
-    public boolean isMultiValue() {
-        return true;
+    protected String valueErrorFor(T value) {
+
+        double number = value.doubleValue();
+
+        if (number > upperLimit.doubleValue() || number < lowerLimit.doubleValue()) {
+            return value + " is out of range " + AbstractNumericProperty.rangeString(lowerLimit, upperLimit);
```

Comment:
Yes that's not great :s Perhaps using something like a decorator would solve that